builders/tools/precompiles/xc20.md

```diff
@@ -0,0 +1,116 @@
+- Create or have two accounts on Moonbase Alpha
+- At least one of the accounts will need to be funded with `DEV` tokens. You can obtain tokens for testing purposes from the Moonbase Alpha [faucet](/builders/get-started/moonbase/#get-tokens/)
+
+## The ERC-20 Interface {: #the-erc20-interface }
+
+The [ERC20.sol](https://github.com/PureStake/moonbeam/blob/master/precompiles/assets-erc20/ERC20.sol) interface on Moonbeam follows the [EIP-20 Token Standard](https://eips.ethereum.org/EIPS/eip-20) which is the standard API interface for tokens within smart contracts. The standard defines the required functions and events that a token contract must implement to be interoperable with different applications.
+
+--8<-- 'text/erc20-interface/erc20-interface.md'
+
+## Retrieve List of Cross-Chain Assets
+
+To fetch a list of the cross-chain assets currently available on the Moonbase Alpha TestNet, head to [Polkadot.js Apps](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fwss.api.moonbase.moonbeam.network#/explorer){target=_blank} and make sure you're connected to Moonbase Alpha. Then click on the **Developer** tab and select **Chain State** from the dropdown. To query the available cross-chain assets, you can follow these steps:
+
+1. From the **selected state query** dropdown, choose **assets**
+2. Select the **asset** extinsic
+3. Disable the **include option** slider
+4. Send the query by clicking on the **+** button
+
+![Fetch list of cross-chain assets](/images/builders/tools/precompiles/xc20/xc20-1.png)
+
+The result will display the asset ID along with some additional information for all of the registered cross-chain assets on Moonbase Alpha. To quickly get more information about a specific asset such as the name and symbol of the asset, you can use the **metadata** extrinsic to return metadata for the asset:
+
+1. From the **selected state query** dropdown, choose **assets**
+2. Select the **metadata** extinsic
+3. Enable the **include option** slider
+4. Enter in the asset ID that was returned from calling the **asset** extrinsic. Please note that if you copy and paste the asset ID with the commas, the commas will automatically be removed and the number might be cut off. Make sure it's the exact same number as the ID. For this example, you can use asset ID `42259045809535163221576417993425387648`
+4. Send the query by clicking on the **+** button 
+
+![Get asset metadata](/images/builders/tools/precompiles/xc20/xc20-2.png)
+
+With the results from the metadata, you can see that the asset ID corresponds to the VUNIT asset.
+
+## Calculate Precompile Addresses
+
+Now that you have retrieved a list of the available cross-chain assets, before you can interact with them via the precompile, you need to derive the precompile address from the asset ID.
+
+The cross-chain asset address is calculated using the following equation:
+
+```
+address = "0XFFFFFFFF" + DecimalToHex(AssetId)
```

Comment:
"this is not always true because the number of FFF can vary, we just need to fill in with F's to reach 40 hex characters"